Auth/auth-service/src/middleware/authorization.ts

```diff
@@ -0,0 +1,31 @@
+import Koa from 'koa';
+import { hasPermissions } from '../utils/permission-util';
+import { decode } from '../utils/token';
+
+const getAccessToken = (header: string | undefined): string | null => {
+    if (!header) return null;
+
+    const [bearer, token] = header.split(' ');
+    return bearer === 'Bearer' && token ? token : null;
+};
+
+export const authorize = (requiredPermissions: string | string[]) => async (
```

Comment:
suggestion: authorize -> isAuthorized and specify a return type such as Promise<boolean>.